src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/EventGridEventExtensions.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Messaging.EventGrid;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+	/// <summary>
+	/// Extension of the EventGridEvent class for creating <see cref="PushNotification"/> object from an <see cref="EventGridEvent "/> object.
+	/// </summary>
+	public static class EventGridEventExtensions
+    {
+		private const string SyncTokenPropertyName = "syncToken";
+
+		/// <summary>
+		/// This method uses an EventGridEvent from EventGrid and tries to create a <see cref="PushNotification"/>
+		/// </summary>
+		/// <param name="eventGridEvent"> EventGridEvent from EventGrid</param>
+		/// <param name="pushNotification"> out parameter set up in this method</param>
+		/// <returns></returns>
+		public static bool TryCreatePushNotification(this EventGridEvent eventGridEvent, out PushNotification pushNotification)
+		{
+			pushNotification = null;
+
+			if (eventGridEvent == null || eventGridEvent.Data == null || eventGridEvent.EventType == null || eventGridEvent.Subject == null)
+			{
+				return false;
+			}
+
+			try
+			{
+				string syncToken = (JsonDocument.Parse(eventGridEvent.Data.ToString()).RootElement)
+					.GetProperty(SyncTokenPropertyName).GetString();
+
+				pushNotification = new PushNotification()
+				{
+					SyncToken = syncToken,
+					EventType = eventGridEvent.EventType,
+					ResourceUri = new Uri(eventGridEvent.Subject)
+				};
+
+				return true;
+			}
+			catch (JsonException) { }
+			catch (ArgumentException) { }
+			catch (InvalidOperationException) { }
+			catch (KeyNotFoundException) { }
```

Comment:
What throws KeyNotFoundException?

Reply:
the `.GetProperty()` 
[Json.GetProperty](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getproperty?view=net-5.0)

Reply:
Is there a ContainsKey method?

Reply:
Looks like there's TryGetProperty and TryGetString.

Reply:
I wasn't able to find TryGetString in the docs? 
Also, its strange that [TryGetProperty]( https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.trygetproperty?view=net-5.0#System_Text_Json_JsonElement_TryGetProperty_System_String_System_Text_Json_JsonElement__) will also throw exceptions.

Reply:
I think I read TryGetSingle too fast.

Reply:
I think we should be good using the TryGetProperty method. It will not throw an exception in our case because our property name is a const string "syncToken" ([code](https://github.com/dotnet/runtime/blob/b07b26dd18ae64cd4a6b091dd4a2c87aee773eec/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs#L237))